src/main/java/games_manager/BoardManagement.java

```diff
@@ -5,7 +5,8 @@
 import java.util.ArrayList;
 
 public interface BoardManagement { //interface that verifies and updates moves on the board
-    public boolean checkLetter(int[] coordinates, String letter, GameBoard board); //returns true if letter is valid
+    public boolean checkLetter(int[] coordinates, String letter, GameBoard board, boolean first_move); //returns true if letter is valid
     public boolean checkWord(ArrayList<MoveInfo> moves, GameBoard board); //returns true if word is valid
     public void updateBoardState(GameBoard board); //updates board after valid turn
+    public GameBoard savePreviousBoardState(GameBoard board); //save the previous board state before move
```

Comment:
Can be removed from the interface and made private because this is a helper function that is not used by any other classes